packages/admin-next/dashboard/src/routes/regions/region-create/components/create-region-form/create-region-form.tsx

```diff
@@ -91,13 +99,22 @@ export const CreateRegionForm = ({
         automatic_taxes: values.automatic_taxes,
       },
       {
-        onSuccess: ({ region }) => {
-          toast.success(t("general.success"), {
-            description: t("regions.toast.create"),
+        onError: (e) => {
```

Comment:
Should we not return on error instead of continuing? No matter what we will reach the success state even if both requests fail.

I also think this being managed by two separate endpoint is a bit unfortunate. Even if two different modules are responsible for the data, could we not manage it simultaneously in our core workflows and HTTP layer, so we don’t have to chain requests?

Reply:
+1

Reply:
That's why I raised the 3rd point, but I think this is a very minor functionality that won't get a lot of interaction, so let's keep it as it is for now. I can create a ticket to revisit this though

Reply:
If not too cumbersome, I would prefer we move it to the workflow sooner rather than later tbh.

Reply:
@sradevski, is this directly related to point 3 thought? Maybe I misunderstood. 

Shouldn't we be able to add this as a workflow step on the BE without addressing what you highlighted in the PR description?

Reply:
Yeah, I think the "issue" here is just the fact that we send two requests to update/created tax-inclusivity on regions. Moving this to a step in the create/updateRegion workflows seems reasonable and should be fairly straightforward.

While as the other question is around enriching currencies and regions responses with the price preferences on the BE, so we don't have to do:
```
const preference = pricePreferences.find(
  (p) => p.attribute === "region_id" && p.value === region.id
)
const preference = pricePreferences.find(
  (p) => p.attribute === "currency_code" && p.value === currency.code
)
```

...right?

I would say the writes are the most sensible to address now, so we get rid of the doubled requests.

Reply:
@olivermrbl yes exactly, to get rid of fetching the preferences in the UI, as they will be part of the region/currency (similar to calculated price in a way, but it will come from the HTTP layer, not a module).

Ok cool, let's do the writes then, and we can leave the reads for now.